File: scratch/replace_gallery.py
import os
import re

photos_js_path = os.path.join(os.path.dirname(__file__), 'photos_js.txt')
root_html_path = os.path.join(os.path.dirname(__file__), '..', 'index.html')
services_html_path = os.path.join(os.path.dirname(__file__), '..', 'servicios el condor', 'index.html')

if not os.path.exists(photos_js_path):
    print("photos_js.txt not found!")
    exit(1)

with open(photos_js_path, 'r', encoding='utf-8') as f:
    photos_js_content = f.read()

def update_html(file_path, prefix=""):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    adjusted_photos = photos_js_content
    if prefix:
        adjusted_photos = photos_js_content.replace("FOTOS/", f"{prefix}FOTOS/")
        
    pattern = r"const photos = \[\s*[\s\S]*?\s*\];"
    replacement = f"const photos = [\n{adjusted_photos}\n        ];"
    
    if not re.search(pattern, content):
        print(f"Pattern not found in {file_path}")
        return
        
    updated_content = re.sub(pattern, replacement, content)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(updated_content)
    print(f"Successfully updated {file_path}")

update_html(root_html_path)
update_html(services_html_path, "../")

File: scratch/replace_gallery.js
const fs = require('fs');
const path = require('path');

const photosJsPath = path.join(__dirname, 'photos_js.txt');
const rootHtmlPath = path.join(__dirname, '..', 'index.html');
const servicesHtmlPath = path.join(__dirname, '..', 'servicios el condor', 'index.html');

if (!fs.existsSync(photosJsPath)) {
    console.error("photos_js.txt not found!");
    process.exit(1);
}

const photosJsContent = fs.readFileSync(photosJsPath, 'utf8');

// Helper to replace the photos array block
function updateHtml(filePath, relativePathPrefix) {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // adjust paths if prefix is provided
    let adjustedPhotosJs = photosJsContent;
    if (relativePathPrefix) {
        adjustedPhotosJs = photosJsContent.replace(/FOTOS\//g, relativePathPrefix + "FOTOS/");
    }
    
    const pattern = /const photos = \[\s*[\s\S]*?\s*\];/;
    const replacement = `const photos = [\n${adjustedPhotosJs}\n        ];`;
    
    if (!pattern.test(content)) {
        console.error(`Pattern not found in ${filePath}`);
        return;
    }
    
    const updatedContent = content.replace(pattern, replacement);
    fs.writeFileSync(filePath, updatedContent, 'utf8');
    console.log(`Successfully updated ${filePath}`);
}

updateHtml(rootHtmlPath, "");
updateHtml(servicesHtmlPath, "../");
